africanmathsinitiative/R-Instat
# Translate 'Code generated by dialog,' comment to the current language (e.g. French)

Fixes part f only of issue #7773 .

@N-thony Please could you peer review?

@rdstern
As requested, I added translation of the 'Code generated by the dialog,' comment. 

It turned out to be more complex than I expected because:
- In the options dialog, the comment can be changed manually by the user, or automatically when the language changes
- In each individual dialog, the comment can be changed manually by the user just for that dialog

So if the language changes, then under which conditions should we replace the current comment with the translated comment? 
I implemented the following approach:

- In the options dialog, if the user changes the language (e.g. to French)
    - If the user has previously manually changed the comment, then the comment is left unchanged 
    - If the user hasn't changed the comment, then the default comment for the new language is used
- In each individual dialog
    - If this is the first time that the dialog opens, then the dialog uses the options dialog comment
    - If the dialog reopens, and the comment has not been manually changed in the dialog, then the dialog uses the options dialog comment
    - If the dialog reopens, and the comment has been manually changed in the dialog, then the dialog uses the manually changed local comment

The general principle is that if the user has manually edited a comment, then this comment will never be overwritten automatically if the language changes. I hope this makes sense.

Please could you test? thanks!

Note: the current French translation of 'Code generated by the dialog,' is just temporary so that you can test this PR. When you approve this PR, I will remove the database change and instead follow the proper CrowdIn translation process via separate PRs. 


### Steps I used to test:

- open Describe / One Variable / Summarise dialog
    - dialog comment should be English default
    - change comment to 'changed in dialog'
    - close dialog
- open Describe / Two/three variables / Describe 2 variables dialog
    - dialog comment should be English default
    - close dialog
- open Tools / Options dialog
    - change language to French
    - instat options comment should change to French
    - click 'OK'
- open Describe / One Variable / Summarize dialog
    - dialog comment should be 'changed in dialog'
    - close dialog
- open Describe / Two/three variables / Describe 2 variables dialog
    - dialog comment should be French default
    - close dialog
- open Describe / General / Column summaries dialog
    - dialog comment should be French default
    - close dialog
- open Tools / Options dialog
    - change language to English
    - instat options comment should change to English
    - click 'OK'
- open Describe / One Variable / Summarise dialog
    - dialog comment should be 'changed in dialog'
    - close dialog
- open Describe / Two/three variables / Describe 2 variables dialog
    - dialog comment should be English default
    - close dialog
- open Describe / Multivariable / Correlation dialog
    - dialog comment should be English default
    - close dialog
